From 27dbf700d68a23f9ea3a11e39666b9979ffa530d Mon Sep 17 00:00:00 2001
From: Emanuele Aina <emanuele.aina@collabora.com>
Date: Sun, 20 Apr 2014 23:58:13 +0200
Subject: [PATCH 08/14] framebuffer: Fix linkage from C++ applications

Prevent Cogl function names from being mangled when a C++ compiler is
being used by adding some missing COGL_{BEGIN,END}_DECLS guards.

https://bugzilla.gnome.org/show_bug.cgi?id=728628

Reviewed-by: Neil Roberts <neil@linux.intel.com>
---
 cogl/deprecated/cogl-framebuffer-deprecated.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cogl/deprecated/cogl-framebuffer-deprecated.h b/cogl/deprecated/cogl-framebuffer-deprecated.h
index 2ac29b1..68ed9d3 100644
--- a/cogl/deprecated/cogl-framebuffer-deprecated.h
+++ b/cogl/deprecated/cogl-framebuffer-deprecated.h
@@ -33,6 +33,8 @@
 
 #include <cogl/cogl-macros.h>
 
+COGL_BEGIN_DECLS
+
 /**
  * cogl_set_framebuffer:
  * @buffer: A #CoglFramebuffer object, either onscreen or offscreen.
@@ -257,4 +259,6 @@ COGL_DEPRECATED_IN_1_18
 CoglPixelFormat
 cogl_framebuffer_get_color_format (CoglFramebuffer *framebuffer);
 
+COGL_END_DECLS
+
 #endif /* __COGL_FRAMEBUFFER_DEPRECATED_H__ */
-- 
2.0.0

